iotileship/iotile/ship/resources/filesystem_manager.py

```diff
@@ -0,0 +1,120 @@
+
+    All Steps using this Resource will use self.root as the root directory to find other files
+
+    Root directory is a pathlib.Path
+
+'''
+
+import subprocess
+import pathlib
+import sys
+
+from iotile.core.utilities.schema_verify import DictionaryVerifier
+from iotile.core.utilities.schema_verify import StringVerifier
+from iotile.core.exceptions import ArgumentError
+from .shared_resource import SharedResource
+
+RESOURCE_ARG_SCHEMA = DictionaryVerifier(desc="filesystem_manager arguments")
+RESOURCE_ARG_SCHEMA.add_required("res_path", StringVerifier("path to resource to work with. "
+                                                            "Specify file:// for a directory or block:// for an unmounted disk partition"))
+
+
+class FilesystemManagerResource(SharedResource):
+    '''
+    Arguments:
+        path (str): Required. One of:
+            file://<path> : Path to a directory that will be worked on
+            block://<path> : Path to a block device to work on (typically /dev/sd...)
+    '''
+
+
+    ARG_SCHEMA = RESOURCE_ARG_SCHEMA
+    _allowed_types = ["file", "block"]
+
+    def __init__(self, args):
+        super(FilesystemManagerResource, self).__init__()
+
+        self._path = args.get('res_path')
+
+        if not "://" in self._path:
+            raise ArgumentError("Cannot detect resource type (looking for something like 'file://)'",
+                                parameter_name="res_path")
+
+        self._type = self._path.split('://')[0]
+        self._resource_loc = self._path.split('://')[1]
+
+        if self._type not in self._allowed_types:
+            raise ArgumentError("FilesystemManagerResource unknown filesystem type", parameter_name="res_path")
+
+        if self._type == "block" and sys.platform != "linux":
+            raise ArgumentError("block;// paths are only supported on Linux. \
+                Mount the device and pass the filepath", parameter_name="res_path")
+
+        if self._type == "block" and sys.platform == "linux":
+            try:
+                subprocess.run(['udisksctl', 'help'], stdout=subprocess.DEVNULL, check=True)
+            except subprocess.CalledProcessError:
+                print("Could not find 'udisksctl'. Make sure it's installed")
+                raise
+
+        self.root = None
+        self._unmount_in_close = False
+
+    #If self._resource_loc is mounted, returns pathlib.Path to mount location
+    #If not, returns None
+    def _get_mountpoint(self):
+        info_cmd = ['udisksctl', 'info', '--block-device', self._resource_loc]
+        info_proc = subprocess.run(info_cmd, stdout=subprocess.PIPE, check=True, universal_newlines=True)
+
+        for line in info_proc.stdout.splitlines():
+            if "MountPoints:" in line:
+                split = line.split()
+                if len(split) > 1:
+                    return pathlib.Path(split[1])
+                return None
+
+        raise RuntimeError("udisksctl failed to report MountPoints")
+
+
```

Comment:
Should be one line between functions in a file, I think. Does your IDE have a PEP8 checker installed? That might make it easier to write stylistically consistent code, but you could also use `pylint` and apply the `.pylintrc` file in the root of coretools.

Reply:
The files I've added are currently clean of pylint issues, though I neglected to go review the styleguide. Will do that.